### ISSUE TITLE
Attempt to use undefined type variable

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#207]
Following code fails. Looks like it is also related to the function being declared after its first usage. 

```fortran
MODULE mod1

LOGICAL :: logext

IMPLICIT NONE

CONTAINS

SUBROUTINE sub1
IMPLICIT NONE
  LOGICAL :: log1
  REAL :: r1

  logext = .TRUE.
  log1 = (rmod() < 2.0) .OR. logext

END SUBROUTINE sub2

ELEMENTAL REAL FUNCTION rmod()
  IMPLICIT NONE
  rmod  =  2.0
END FUNCTION rmod

END MODULE mod1
```

```bash
"mod1.f90", line 24: attempt to use undefined type variable, logext
```

Type of `logext ` is clear, it is a `LOGICAL`

**Comment by clementval** [comment]
PR waiting for merge #210

**Comment by clementval** [comment]
Closed with #210
